cmd/proxy/main.go

```diff
@@ -32,6 +33,9 @@ func main() {
 	if err != nil {
 		log.Fatalf("could not load config file: %v", err)
 	}
+	if err := conf.Validate(errors.Op("main")); err != nil {
```

Comment:
We already have a `validateConfig` function here: https://github.com/gomods/athens/blob/master/pkg/config/config.go#L238

It gets automatically called when you call `config.Load` 

So this is a bit confusing :)

In the same fashion, the `mode.Validate` function in this case is getting called twice, once here and once in `mode.NewFile`.

I think we can just:
1. remove this line here 
2. remove `config.Validate` function 
3. make `mode.Validate` private

Reply:
@marwan-at-work I added a `Validate` function on config structs so that we can nest validations. I'll try to move all of the logic into [`validate`](https://pkg.go.dev/gopkg.in/go-playground/validator.v9) struct tags though